components/CustomerForm.tsx

```diff
@@ -33,25 +120,119 @@ class CustomerForm extends React.Component<{
     )
   }
 
-  render() {
+  if (createCustomerResponse) {
+    const copyButton = (
+      <CopyButton value={createCustomerResponse.email}>
+        {({ copied, copy }) => (
+          <ActionIcon color={copied ? 'green' : 'gray'} onClick={copy}>
+            {copied ? <IconClipboardCheck /> : <IconClipboardCopy />}
+          </ActionIcon>
+        )}
+      </CopyButton>
+    )
+
+    const completedText = appData.accessToken ? (
+      <Group spacing={5}>
+        <IconPlugConnected />
+        You are now linked to Slope as <Code>{createCustomerResponse.email}</Code>
+        {copyButton} password:
+        <Code>{createCustomerResponse.password}</Code>
+      </Group>
+    ) : (
+      <Group spacing={5}>
+        <IconUserPlus />
+        Created customer <Code>{createCustomerResponse.email}</Code>
+        {copyButton} use password:
+        <Code>{createCustomerResponse.password}</Code>
+      </Group>
+    )
+
     return (
-      <>
-        {this.renderFormField('email', 'Email')}
-        <SimpleGrid cols={2}>
-          {this.renderFormField('businessName', 'Business Name')}
-          {this.renderFormField('phone', 'Phone')}
-        </SimpleGrid>
-        <SimpleGrid cols={2}>
-          {this.renderFormField('line1', 'Address')}
-          {this.renderFormField('city', 'City')}
-        </SimpleGrid>
-        <SimpleGrid cols={2}>
-          {this.renderFormField('state', 'State')}
-          {this.renderFormField('postalCode', 'Postal Code')}
-        </SimpleGrid>
-      </>
+      <Box bg="fog.3">
+        <Container>
+          <Center>
+            <Text py="md" color="black" fw={600} size="sm">
+              {completedText}
+            </Text>
+          </Center>
+        </Container>
+      </Box>
     )
   }
+
+  return (
+    <Box bg="gray.1">
```

Comment:
spread rest of props on root rendered component? generally a great practice anywhere you go

```tsx
// in props destructure
{ appData, updateAppData, ...boxProps }: { appData: any, updateAppData: () => void } & BoxProps }
```

```suggestion
    <Box bg="gray.1" {...boxProps}>
```

Reply:
will do